tests/test_unittest/test_gtc/test_npir_gen.py

```diff
@@ -0,0 +1,269 @@
+import re
+from typing import Iterator, Set
+
+import pytest
+
+from gt4py.gtc import common
+from gt4py.gtc.python import npir, npir_gen
+
+
+UNDEFINED_DTYPES = {common.DataType.INVALID, common.DataType.AUTO, common.DataType.DEFAULT}
+
+DEFINED_DTYPES: Set[common.DataType] = set(common.DataType) - UNDEFINED_DTYPES  # type: ignore
+
+
+class FieldSliceBuilder:
+    def __init__(self, name: str, *, parallel_k=False):
+        self._name = name
+        self._offsets = [0, 0, 0]
+        self._parallel_k = parallel_k
+
+    def offsets(self, i: int, j: int, k: int):
+        self._offsets = [i, j, k]
+        return self
+
+    def build(self):
+        return npir.FieldSlice(
+            name=self._name,
+            i_offset=npir.AxisOffset.i(self._offsets[0]),
+            j_offset=npir.AxisOffset.j(self._offsets[1]),
+            k_offset=npir.AxisOffset.k(self._offsets[2], parallel=self._parallel_k),
+        )
+
+
+@pytest.fixture(params=DEFINED_DTYPES)
+def defined_dtype(request) -> Iterator[common.DataType]:
+    yield request.param
+
+
+def test_literal(defined_dtype: common.DataType) -> None:
+    result = npir_gen.NpirGen.apply(npir.Literal(dtype=defined_dtype, value="42"))
```

Comment:
I think I would use
```suggestion
    result = npir_gen.NpirGen().visit(npir.Literal(dtype=defined_dtype, value="42"))
```
and restrict the apply to the root node, to allow the apply to do some special stuff for the root node if necessary. Not sure if it makes sense.

Reply:
If that is a pattern used in any code generator we should stick to it even where we don't use it.